src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -312,15 +312,15 @@ object Task {
   object Id {
 
     object Names {
-      val anonymousContainer = "$anon" // presence of `$` is important since it's illegal for a real container name!
+      val anonymousContainer = "_anon" // presence of `_` is important since it's illegal for a real container name!
```

Comment:
I agree with @alenkacz, `_app` is more expressive.

```suggestion
      val anonymousContainer = "_app" // presence of `_` is important since it's illegal for a real container name!
```